libsolidity/codegen/CompilerUtils.cpp

```diff
@@ -1266,6 +1271,10 @@ void CompilerUtils::rightShiftNumberOnStack(unsigned _bits)
 
 unsigned CompilerUtils::prepareMemoryStore(Type const& _type, bool _padToWords)
 {
+	solUnimplementedAssert(_type.canBeStored(),
```

Comment:
The documentation of `canBeStored` is talking about storage - this function handles memory. Please add an assertion that type is a value type as mentioned in the issue description.